### PR TITLE
Lodash is no need

### PR DESCRIPTION
**Pull request by ftdebugger**

### CODE REVIEW

**Comment by ftdebugger** [comment]
@next0, can you review this?

File: index.js [hunk 1]
```diff
@@ -37,9 +24,14 @@ omitBy = omitBy || omit;
  * @returns {Array}
  */
 let map = (object, callback) => {
-    return Object.keys(object).map(key => {
-        return callback(object[key], key);
-    });
+    let keys = Object.keys(object),
```
Comment (next0) [inline index.js]: `let objKeys = keys(object);`

File: index.js [hunk 2]
```diff
@@ -63,24 +57,163 @@ let forEach = (object, callback) => {
...
+};
 
 /**
- * @typedef {{session: function, put: function}} DiContainer
+ * @param {{}} object
+ *
+ * @returns {string[]}
  */
+let functions = (object) => {
+    return Object.keys(object).filter(key => isFunction(object[key]));
```
Comment (next0) [inline index.js]: `return keys(object).filter(key => isFunction(object[key]));`

File: index.js [hunk 3]
```diff
@@ -63,24 +57,163 @@ let forEach = (object, callback) => {
...
+ * @returns {{}}
+ */
+let clone = (object) => {
+    return extend({}, object);
+};
+
+/**
+ * @type {function(arr: *): boolean}
+ */
+let isArray = Array.isArray ? Array.isArray : arr => arr instanceof Array;
```
Comment (next0) [inline index.js:218]: `arr => arr instanceof Array` может как и с `isFunction` проверить как `(array) => Object.prototype.toString.call(array) === '[object Array]';`

**Pull request merged**